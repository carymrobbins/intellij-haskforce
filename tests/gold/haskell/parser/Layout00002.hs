module Layout00002 where

instance Monad (DocM s) where
        return = retDocM

{-# INLINE thenDocM #-}
{-# INLINE then_DocM #-}

thenDocM :: DocM s a -> (a -> DocM s b) -> DocM s b

