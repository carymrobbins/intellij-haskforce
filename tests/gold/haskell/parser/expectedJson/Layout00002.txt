[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":10,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":10,"srcSpanEndLine":10,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":20}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":8},"srcInfoPoints":[]},"Layout00002"]},null,null]},[],[],[{"InstDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":3,"srcSpanEndLine":6,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":25},{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":9},{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1}]},null,{"IHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":10},"srcInfoPoints":[]},"Monad"]}]},[{"TyParen":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":16},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":16},{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":23}]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17},"srcInfoPoints":[]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17},"srcInfoPoints":[]},"DocM"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":22},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":22},"srcInfoPoints":[]},"s"]}]}]}]}]]},[{"InsDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":9},"srcInfoPoints":[]},{"PatBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":9},"srcInfoPoints":[]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":9},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":9},"srcInfoPoints":[]},"return"]}]},null,{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":16},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":16}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":18},"srcInfoPoints":[]},"retDocM"]}]}]}]},null]}]}]]},{"InlineSig":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":21}]},true,null,{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":12},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":12},"srcInfoPoints":[]},"thenDocM"]}]}]},{"InlineSig":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":22}]},true,null,{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":12},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":12},"srcInfoPoints":[]},"then_DocM"]}]}]},{"TypeSig":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":52,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":10}]},[{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":1},"srcInfoPoints":[]},"thenDocM"]}],{"TyFun":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":52,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":13},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":22}]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":13},"srcInfoPoints":[]},"DocM"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":18},"srcInfoPoints":[]},"s"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":20},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":20},"srcInfoPoints":[]},"a"]}]}]},{"TyFun":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":52,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":25},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":43,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":41}]},{"TyParen":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":25},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":25},{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":39}]},{"TyFun":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":39,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":26},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":28}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":26},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":26},"srcInfoPoints":[]},"a"]}]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":39,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":31},"srcInfoPoints":[]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":37,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":31},"srcInfoPoints":[]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":31},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":31},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":31},"srcInfoPoints":[]},"DocM"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":37,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":36},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":37,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":36},"srcInfoPoints":[]},"s"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":39,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":38},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":39,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":38},"srcInfoPoints":[]},"b"]}]}]}]}]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":52,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":44},"srcInfoPoints":[]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":50,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":44},"srcInfoPoints":[]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":48,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":44},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":48,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":44},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":48,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":44},"srcInfoPoints":[]},"DocM"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":50,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":49},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":50,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":49},"srcInfoPoints":[]},"s"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":52,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":51},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":52,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":51},"srcInfoPoints":[]},"b"]}]}]}]}]}]}]]},[]]