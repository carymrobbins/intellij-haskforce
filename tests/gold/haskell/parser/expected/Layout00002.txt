Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('Layout00002')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    INSTANCEDECL
      PsiElement(HaskellTokenType.instance)('instance')
      CTYPE
        TYPEE
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('Monad')
          ATYPE
            PsiElement(HaskellTokenType.()('(')
            CTYPE
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('DocM')
                ATYPE
                  TYVAR
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('s')
            PsiElement(HaskellTokenType.))(')')
      PsiElement(HaskellTokenType.where)('where')
      IDECL
        FUNORPATDECL
          PAT
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('return')
          RHS
            PsiElement(HaskellTokenType.=)('=')
            EXP
              QVAR
                QVARID
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('retDocM')
      PPRAGMA
        PsiElement(HaskellTokenType.{-#)('{-#')
        PsiElement(HaskellTokenType.PRAGMA)('INLINE')
        PsiElement(HaskellTokenType.PRAGMA)('thenDocM')
        PsiElement(HaskellTokenType.#-})('#-}')
      PPRAGMA
        PsiElement(HaskellTokenType.{-#)('{-#')
        PsiElement(HaskellTokenType.PRAGMA)('INLINE')
        PsiElement(HaskellTokenType.PRAGMA)('then_DocM')
        PsiElement(HaskellTokenType.#-})('#-}')
    GENDECL
      VARS
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('thenDocM')
      PsiElement(HaskellTokenType.::)('::')
      CTYPE
        TYPEE
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('DocM')
          ATYPE
            TYVAR
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('s')
          ATYPE
            TYVAR
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('a')
          PsiElement(HaskellTokenType.->)('->')
          TYPEE
            ATYPE
              PsiElement(HaskellTokenType.()('(')
              CTYPE
                TYPEE
                  ATYPE
                    TYVAR
                      VARID
                        PsiElement(HaskellTokenType.varidRegexp)('a')
                  PsiElement(HaskellTokenType.->)('->')
                  TYPEE
                    ATYPE
                      OQTYCON
                        QTYCON
                          TYCON
                            CONID
                              PsiElement(HaskellTokenType.conidRegexp)('DocM')
                    ATYPE
                      TYVAR
                        VARID
                          PsiElement(HaskellTokenType.varidRegexp)('s')
                    ATYPE
                      TYVAR
                        VARID
                          PsiElement(HaskellTokenType.varidRegexp)('b')
              PsiElement(HaskellTokenType.))(')')
            PsiElement(HaskellTokenType.->)('->')
            TYPEE
              ATYPE
                OQTYCON
                  QTYCON
                    TYCON
                      CONID
                        PsiElement(HaskellTokenType.conidRegexp)('DocM')
              ATYPE
                TYVAR
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('s')
              ATYPE
                TYVAR
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('b')